server/rpc/SessionRPC.java

```diff
@@ -40,23 +39,23 @@
 
     private final Grakn.Session session;
     private final GraknRPCService graknRPCService;
-    private final Set<TransactionRPC> transactionRPCs;
+    private final ConcurrentHashMap<Integer, TransactionRPC> transactionRPCs;
```

Comment:
Intermittently, clients would stall when attempting to close many transactions in parallel. Transactions were stored in a non-threadsafe collection (a `HashSet`) on the server, which was being modified concurrently by multiple threads. Sometimes, this caused iteration over the collection to not return all the elements, which meant the missed Transactions would remain open indefinitely, causing the client to stall.

We changed the server implementation to use a `ConcurrentHashMap` to remedy the issue.